### PR TITLE
feat(argo-cd): Add support for aggregation clusterroles

### PR DESCRIPTION
**Pull request by abyth**
* feat: Add support for aggregation clusterroles

Signed-off-by: Christian Strack <github@abyth.de>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [X] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [X] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [X] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/README.md [hunk 1]
```diff
@@ -147,7 +147,7 @@ NAME: my-release
 | nameOverride | Provide a name in place of `argocd` | `"argocd"` |
 | fullnameOverride | String to fully override `"argo-cd.fullname"` | `""` |
 | apiVersionOverrides.certmanager | String to override apiVersion of certmanager resources rendered by this helm chart | `""` |
-| apiVersionOverrides.ingress | String to override apiVersion of ingresses rendered by this helm chart | `""` |
+| apiVersionOverrides.ingress | String to override apiVersion of ingresses rendered by this helm chart | `""` |
```
Comment (mbevc1) [inline charts/argo-cd/README.md:150]:
```suggestion
| apiVersionOverrides.ingress | String to override apiVersion of ingresses rendered by this helm chart | `""` |
```

File: charts/argo-cd/templates/argocd-aggregate-roles.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{{- if .Values.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "argo-cd.fullname" . }}-aggregate-to-view
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
+    {{- include "argo-cd.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-aggregate-roles.yaml:8]: I think we do not need `component` and `name` labels set for these resources.
```suggestion
    {{- include "argo-cd.labels" (dict "context" .) | nindent 4 }}
```

File: charts/argo-cd/templates/argocd-aggregate-roles.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  - watch
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "argo-cd.fullname" . }}-aggregate-to-edit
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-edit: "true"
+    {{- include "argo-cd.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-aggregate-roles.yaml:27]:
```suggestion
    {{- include "argo-cd.labels" (dict "context" .) | nindent 4 }}
```

File: charts/argo-cd/templates/argocd-aggregate-roles.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  - watch
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "argo-cd.fullname" . }}-aggregate-to-admin
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
+    {{- include "argo-cd.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-aggregate-roles.yaml:51]:
```suggestion
    {{- include "argo-cd.labels" (dict "context" .) | nindent 4 }}
```

File: charts/argo-cd/templates/argocd-aggregate-roles.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
+{{- if .Values.rbac.create }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-aggregate-roles.yaml:1]: I am unsure if we sould use this generic key for this :-)
Normally when setting `rbac.create` on a chart, the chart would not deploy rbac resources at all.

File: charts/argo-cd/templates/argocd-aggregate-roles.yaml (hunk 2 above)
Comment (abyth) [reply charts/argo-cd/templates/argocd-aggregate-roles.yaml:8]: I was not sure how people are using existing labels and therefore decided to align on the existing ones. I'm fine with the removal

File: charts/argo-cd/templates/argocd-aggregate-roles.yaml (hunk 5 above)
Comment (abyth) [reply charts/argo-cd/templates/argocd-aggregate-roles.yaml:1]: I get your point.. i spent quite some time to look up how other charts are dealing with that behaviour. E.g. `argo-rollouts` is using the `clusterInstall` flag for it, `cert-manager` uses `rbac`. I agree that this term is potentially misleading. Do you have a suggestions?

Comment (mkilchhofer) [reply charts/argo-cd/templates/argocd-aggregate-roles.yaml:1]: IDK what would fit perfectly. But I spottet that we use `createAggregateRoles: true` in argo-workflows: https://github.com/argoproj/argo-helm/blob/master/charts/argo-workflows/values.yaml#L8

File: charts/argo-cd/Chart.yaml [hunk 6]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: 2.0.5
 description: A Helm chart for ArgoCD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 3.11.5
+version: 3.11.6
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:5]: As this is a feature, please bump the minor version :-)

File: charts/argo-cd/templates/argocd-aggregate-roles.yaml (hunk 5 above)
Comment (abyth) [reply charts/argo-cd/templates/argocd-aggregate-roles.yaml:1]: Sounds good

File: charts/argo-cd/Chart.yaml (hunk 6 above)
Comment (abyth) [reply charts/argo-cd/Chart.yaml:5]: My bad..

**Review by mkilchhofer — APPROVED** [review]
LGTM

**Pull request merged**